### PR TITLE
Add config for shadow fields

### PR DESCRIPTION
**Pull request by nkylstad**
## Description

Update to DataType model for configuration of shadow fields in applicationMetadata.

## Related Issue(s)
- https://github.com/Altinn/altinn-studio/issues/10108

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Storage.Interface/Models/ShadowFields.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        /// </summary>
+        [JsonProperty(PropertyName = "prefix")]
+        public string Prefix { get; set; }
+
+        /// <summary>
+        /// Gets or sets the data type to save filtered data (without shadow fields) to.
+        /// Optional. If not set, the containing data type will be updated.
+        /// </summary>
+        [JsonProperty(PropertyName = "saveToDataType")]
+        public string SaveToDataType { get; set; }
```
Comment (acn-sbuad) [inline src/Storage.Interface/Models/ShadowFields.cs:22]: Wondering why we need a reference to the dataType withing the shadowField object if it is already a sub-property of a given data type?

Comment (nkylstad) [reply src/Storage.Interface/Models/ShadowFields.cs:22]: The property is meant to specify the data type to save the shadow fields to - i.e. a different data type. If specified, we will use this dataType and save the updated data to that dataType. If not specified, we use the containing data type (the one that this config is set for)

Comment (nkylstad) [reply src/Storage.Interface/Models/ShadowFields.cs:22]: Would something like `DataTypeForFilteredData` be clearer?

**Review by acn-sbuad — APPROVED** [review]
🎈

File: src/Storage.Interface/Models/ShadowFields.cs (hunk 1 above)
Comment (acn-sbuad) [reply src/Storage.Interface/Models/ShadowFields.cs:22]: I think we can leave it as is. Based on your

**Pull request merged**